In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [12]:
df = pd.read_csv('./cancer_reg.csv')

In [9]:
df.head()

avgAnnCount  avgDeathsPerYear  TARGET_deathRate  incidenceRate  medIncome  \
0       1397.0               469             164.9          489.8      61898   
1        173.0                70             161.3          411.6      48127   
2        102.0                50             174.7          349.7      49348   
3        427.0               202             194.8          430.4      44243   
4         57.0                26             144.4          350.1      49955   

   popEst2015  povertyPercent  studyPerCap           binnedInc  MedianAge  \
0      260131            11.2   499.748204   (61494.5, 125635]       39.3   
1       43269            18.6    23.111234  (48021.6, 51046.4]       33.0   
2       21026            14.6    47.560164  (48021.6, 51046.4]       45.0   
3       75882            17.1   342.637253    (42724.4, 45201]       42.8   
4       10321            12.5     0.000000  (48021.6, 51046.4]       48.3   

   ...  PctPrivateCoverageAlone  PctEmpPrivCoverage PctPublicCoverage  \
0  ...                      NaN                41.6              32.9   
1  ...                     53.8                43.6              31.1   
2  ...                     43.5                34.9              42.1   
3  ...                     40.3                35.0              45.3   
4  ...                     43.9                35.1              44.0   

   PctPublicCoverageAlone   PctWhite  PctBlack  PctAsian  PctOtherRace  \
0                    14.0  81.780529  2.594728  4.821857      1.843479   
1                    15.3  89.228509  0.969102  2.246233      3.741352   
2                    21.1  90.922190  0.739673  0.465898      2.747358   
3                    25.0  91.744686  0.782626  1.161359      1.362643   
4                    22.7  94.104024  0.270192  0.665830      0.492135   

   PctMarriedHouseholds  BirthRate  
0             52.856076   6.118831  
1             45.372500   4.333096  
2             54.444868   3.729488  
3             51.021514   4.603841  
4             54.027460   6.796657  

[5 rows x 34 columns]

In [29]:
df.Geography

0          Kitsap County, Washington
1        Kittitas County, Washington
2       Klickitat County, Washington
3           Lewis County, Washington
4         Lincoln County, Washington
                    ...             
3042        Ellsworth County, Kansas
3043           Finney County, Kansas
3044             Ford County, Kansas
3045         Franklin County, Kansas
3046            Geary County, Kansas
Name: Geography, Length: 3047, dtype: object

In [13]:
df.columns

Index(['avgAnnCount', 'avgDeathsPerYear', 'TARGET_deathRate', 'incidenceRate',
       'medIncome', 'popEst2015', 'povertyPercent', 'studyPerCap', 'binnedInc',
       'MedianAge', 'MedianAgeMale', 'MedianAgeFemale', 'Geography',
       'AvgHouseholdSize', 'PercentMarried', 'PctNoHS18_24', 'PctHS18_24',
       'PctSomeCol18_24', 'PctBachDeg18_24', 'PctHS25_Over',
       'PctBachDeg25_Over', 'PctEmployed16_Over', 'PctUnemployed16_Over',
       'PctPrivateCoverage', 'PctPrivateCoverageAlone', 'PctEmpPrivCoverage',
       'PctPublicCoverage', 'PctPublicCoverageAlone', 'PctWhite', 'PctBlack',
       'PctAsian', 'PctOtherRace', 'PctMarriedHouseholds', 'BirthRate'],
      dtype='object')

In [4]:
df['avgAnnCount'].count()

3047

We can see below that there are no null values in the avgAnnCount column. We can conclude there are 3,047 total rows in our data.

In [11]:
#check data quality
#identify rows with null values
df.isnull().sum()

avgAnnCount                   0
avgDeathsPerYear              0
TARGET_deathRate              0
incidenceRate                 0
medIncome                     0
popEst2015                    0
povertyPercent                0
studyPerCap                   0
binnedInc                     0
MedianAge                     0
MedianAgeMale                 0
MedianAgeFemale               0
Geography                     0
AvgHouseholdSize              0
PercentMarried                0
PctNoHS18_24                  0
PctHS18_24                    0
PctSomeCol18_24            2285
PctBachDeg18_24               0
PctHS25_Over                  0
PctBachDeg25_Over             0
PctEmployed16_Over          152
PctUnemployed16_Over          0
PctPrivateCoverage            0
PctPrivateCoverageAlone     609
PctEmpPrivCoverage            0
PctPublicCoverage             0
PctPublicCoverageAlone        0
PctWhite                      0
PctBlack                      0
PctAsian                      0
PctOther

In [7]:
df['PctSomeCol18_24'].describe()

count    762.000000
mean      40.977034
std       11.115805
min        7.100000
25%       34.000000
50%       40.400000
75%       46.400000
max       79.000000
Name: PctSomeCol18_24, dtype: float64

The 'PctSomeCol18_24' represents the percent of county residents ages 18-24 whose highest education is some college. We are missing about 70% of the data for this column. Furthermore, we have complete data for residents who have completed high school and college. Therfore, it makes more sense to drop this column than attempt to fill it with an educated guess.

In [14]:
df.drop(columns = ['PctSomeCol18_24'], inplace = True)

In [16]:
df['PctPrivateCoverageAlone'].describe()

count    2438.000000
mean       48.453774
std        10.083006
min        15.700000
25%        41.000000
50%        48.700000
75%        55.600000
max        78.900000
Name: PctPrivateCoverageAlone, dtype: float64

Similarly with 'PctPrivateCoverageAlone', we have a lot of good indicators regarding health insurance such as percentages covered by employers, publicly, and privately. There is also a fair amount of variability in the values present for percent private alone. Therefore, I will also drop this column and use the more reliable indicators we have present.

In [17]:
df.drop(columns = ['PctPrivateCoverageAlone'], inplace = True)

In [20]:
null= df[df['PctEmployed16_Over'].isnull()]

In [22]:
#view stats of the data set with remaining null values
null.describe()

avgAnnCount  avgDeathsPerYear  TARGET_deathRate  incidenceRate  \
count   152.000000        152.000000        152.000000     152.000000   
mean    513.028582        157.480263        179.142105     453.991077   
std     767.042062        270.591943         32.836207      81.917739   
min      12.000000          4.000000        102.600000     254.700000   
25%      80.250000         28.750000        159.075000     423.875000   
50%     171.000000         62.000000        179.400000     453.549422   
75%     581.250000        169.250000        195.525000     483.125000   
max    5978.000000       2528.000000        362.800000    1206.900000   

           medIncome     popEst2015  povertyPercent  studyPerCap   MedianAge  \
count     152.000000     152.000000      152.000000   152.000000  152.000000   
mean    48232.473684   82806.506579       16.168421   111.731619   48.471053   
std     11882.914554  132219.220386        6.067342   207.586783   58.715825   
min     28716.000000    1518.000000        4.600000     0.000000   24.400000   
25%     40687.250000   13137.250000       12.175000     0.000000   37.650000   
50%     45988.000000   28176.500000       15.100000     0.000000   40.800000   
75%     52668.750000   96009.750000       18.825000   152.373117   43.800000   
max    106871.000000  949827.000000       34.000000   837.122269  511.200000   

       MedianAgeMale  ...  PctPrivateCoverage  PctEmpPrivCoverage  \
count     152.000000  ...          152.000000          152.000000   
mean       39.048684  ...           65.301974           41.977632   
std         5.102460  ...           10.660986            9.607849   
min        24.300000  ...           29.600000           18.400000   
25%        36.475000  ...           58.700000           35.675000   
50%        39.400000  ...           66.150000           42.050000   
75%        42.225000  ...           73.200000           48.500000   
max        51.600000  ...           85.700000           65.100000   

       PctPublicCoverage  PctPublicCoverageAlone    PctWhite    PctBlack  \
count         152.000000              152.000000  152.000000  152.000000   
mean           35.084868               18.603289   84.304434    8.614817   
std             7.848447                6.039872   16.930060   14.174153   
min            17.000000                8.200000   14.086758    0.000000   
25%            31.175000               13.775000   78.484248    0.563287   
50%            35.300000               18.300000   91.946716    1.294100   
75%            39.725000               22.425000   96.007865    9.349367   
max            57.900000               38.000000   99.498433   59.538990   

         PctAsian  PctOtherRace  PctMarriedHouseholds   BirthRate  
count  152.000000    152.000000            152.000000  152.000000  
mean     1.125967      1.557219             51.566497    5.705255  
std      1.734988      2.316574              7.482360    1.663989  
min      0.000000      0.000000             26.746115    0.569801  
25%      0.313271      0.283617             47.154062    4.628424  
50%      0.624812      0.788103             51.490974    5.325615  
75%      1.206910      1.917561             56.406664    6.484724  
max     16.240896     15.653850             78.075397   12.219731  

[8 rows x 30 columns]

In [23]:
#view stats of the data set as a whole
df.describe()

avgAnnCount  avgDeathsPerYear  TARGET_deathRate  incidenceRate  \
count   3047.000000       3047.000000       3047.000000    3047.000000   
mean     606.338544        185.965868        178.664063     448.268586   
std     1416.356223        504.134286         27.751511      54.560733   
min        6.000000          3.000000         59.700000     201.300000   
25%       76.000000         28.000000        161.200000     420.300000   
50%      171.000000         61.000000        178.100000     453.549422   
75%      518.000000        149.000000        195.200000     480.850000   
max    38150.000000      14010.000000        362.800000    1206.900000   

           medIncome    popEst2015  povertyPercent  studyPerCap    MedianAge  \
count    3047.000000  3.047000e+03     3047.000000  3047.000000  3047.000000   
mean    47063.281917  1.026374e+05       16.878175   155.399415    45.272333   
std     12040.090836  3.290592e+05        6.409087   529.628366    45.304480   
min     22640.000000  8.270000e+02        3.200000     0.000000    22.300000   
25%     38882.500000  1.168400e+04       12.150000     0.000000    37.700000   
50%     45207.000000  2.664300e+04       15.900000     0.000000    41.000000   
75%     52492.000000  6.867100e+04       20.400000    83.650776    44.000000   
max    125635.000000  1.017029e+07       47.400000  9762.308998   624.000000   

       MedianAgeMale  ...  PctPrivateCoverage  PctEmpPrivCoverage  \
count    3047.000000  ...         3047.000000         3047.000000   
mean       39.570725  ...           64.354939           41.196324   
std         5.226017  ...           10.647057            9.447687   
min        22.400000  ...           22.300000           13.500000   
25%        36.350000  ...           57.200000           34.500000   
50%        39.600000  ...           65.100000           41.100000   
75%        42.500000  ...           72.100000           47.700000   
max        64.700000  ...           92.300000           70.700000   

       PctPublicCoverage  PctPublicCoverageAlone     PctWhite     PctBlack  \
count        3047.000000             3047.000000  3047.000000  3047.000000   
mean           36.252642               19.240072    83.645286     9.107978   
std             7.841741                6.113041    16.380025    14.534538   
min            11.200000                2.600000    10.199155     0.000000   
25%            30.900000               14.850000    77.296180     0.620675   
50%            36.300000               18.800000    90.059774     2.247576   
75%            41.550000               23.100000    95.451693    10.509732   
max            65.100000               46.600000   100.000000    85.947799   

          PctAsian  PctOtherRace  PctMarriedHouseholds    BirthRate  
count  3047.000000   3047.000000           3047.000000  3047.000000  
mean      1.253965      1.983523             51.243872     5.640306  
std       2.610276      3.517710              6.572814     1.985816  
min       0.000000      0.000000             22.992490     0.000000  
25%       0.254199      0.295172             47.763063     4.521419  
50%       0.549812      0.826185             51.669941     5.381478  
75%       1.221037      2.177960             55.395132     6.493677  
max      42.619425     41.930251             78.075397    21.326165  

[8 rows x 30 columns]

When we compare the summmary statistics of the larger data set with the data set containing only the remaining null values, we see that the two sets are very similar.

Employment rate could be a relevant economic indicator that could have ties to things like receiving medical treatment and, therefore, being diagnosed. So I take this to be relevant, though certainly not a decisive factor. Rather that lose the 80% of the data points we have, I will replace null values with a mean or median value of the column. The mean and median are only .35% apart, and the data in the column are close to normally distributed. Therefore, the mean seems like the best replacement option.

In [24]:
df['PctEmployed16_Over'].describe()

count    2895.000000
mean       54.152642
std         8.315064
min        17.600000
25%        48.600000
50%        54.500000
75%        60.300000
max        80.100000
Name: PctEmployed16_Over, dtype: float64

In [26]:
df['PctEmployed16_Over'].median()

54.5

In [27]:
#fill remaining null values with the mean value of the column
df['PctEmployed16_Over'].fillna(df['PctEmployed16_Over'].mean(), inplace= True)

In [28]:
#verify no remaining null values
df.isnull().sum()

avgAnnCount               0
avgDeathsPerYear          0
TARGET_deathRate          0
incidenceRate             0
medIncome                 0
popEst2015                0
povertyPercent            0
studyPerCap               0
binnedInc                 0
MedianAge                 0
MedianAgeMale             0
MedianAgeFemale           0
Geography                 0
AvgHouseholdSize          0
PercentMarried            0
PctNoHS18_24              0
PctHS18_24                0
PctBachDeg18_24           0
PctHS25_Over              0
PctBachDeg25_Over         0
PctEmployed16_Over        0
PctUnemployed16_Over      0
PctPrivateCoverage        0
PctEmpPrivCoverage        0
PctPublicCoverage         0
PctPublicCoverageAlone    0
PctWhite                  0
PctBlack                  0
PctAsian                  0
PctOtherRace              0
PctMarriedHouseholds      0
BirthRate                 0
dtype: int64

Before the dataframe is exported, I am going to drop unwanted columns. In this case, the column marked 'TARGET' is not what I am interested in identifying so I will drop it. Additionally, 'incidencerate' is a calculation of the mean number of diagnoses divided by 100,000, and assumes we already have a tally for the cancer rate - i.e. the variable we are trying to estimate. We need to drop it as well.

In [31]:
df.drop(columns = ['TARGET_deathRate', 'incidenceRate'], inplace = True)

In [33]:
#export tidy dataframe
df.to_csv('./df.csv', index= False)